packages/js/src/feeds/feeds.ts

```diff
@@ -1,33 +1,187 @@
+    return this.callWithSession(async () => {
+      const args = { feedIdentifier, status };
+      try {
+        this._emitter.emit('feeds.fetch_count.pending', { args });
+
+        let response: { count: number };
+        if (SEEN_OR_UNSEEN.includes(status)) {
+          response = await this._apiService.getUnseenCount({
+            feedIdentifier,
+            seen: status === NotificationStatus.SEEN,
+          });
+        } else {
+          response = await this._apiService.getUnreadCount({
+            feedIdentifier,
+            read: status === NotificationStatus.READ,
+          });
+        }
+
+        this._emitter.emit('feeds.fetch_count.success', { args, result: response.count });
+
+        return response.count;
+      } catch (error) {
+        this._emitter.emit('feeds.fetch_count.error', { args, error });
+        throw error;
+      }
+    });
+  }
+
+  async markNotificationAs(args: MarkNotificationAsByIdArgs): Promise<Notification>;
+  async markNotificationAs(args: MarkNotificationAsByInstanceArgs): Promise<Notification>;
+  async markNotificationAs(args: MarkNotificationAsArgs): Promise<Notification> {
+    return this.callWithSession(async () =>
+      markNotificationAs({
+        emitter: this._emitter,
+        apiService: this._apiService,
+        args,
+      })
+    );
+  }
+
+  async markNotificationsAs(args: MarkNotificationsAsByIdsArgs): Promise<Notification[]>;
+  async markNotificationsAs(args: MarkNotificationsAsByInstancesArgs): Promise<Notification[]>;
+  async markNotificationsAs(args: MarkNotificationsAsArgs): Promise<Notification[]> {
+    return this.callWithSession(async () =>
+      markNotificationsAs({
+        apiService: this._apiService,
+        emitter: this._emitter,
+        args,
+      })
+    );
+  }
+
+  async markAllNotificationsAs({
+    feedIdentifier,
+    status = NotificationStatus.SEEN,
+  }: MarkAllNotificationsAsArgs): Promise<number> {
+    return this.callWithSession(async () => {
+      const args = { feedIdentifier, status };
+      try {
+        this._emitter.emit('feeds.mark_all_notifications_as.pending', { args });
+
+        let response = 0;
+        if (status === NotificationStatus.SEEN) {
+          response = await this._apiService.markAllMessagesAsSeen(feedIdentifier);
+        } else {
+          response = await this._apiService.markAllMessagesAsRead(feedIdentifier);
+        }
+
+        this._emitter.emit('feeds.mark_all_notifications_as.success', {
+          args,
+          result: response,
+        });
 
         return response;
       } catch (error) {
-        this._emitter.emit('feeds.fetch.error', { error });
+        this._emitter.emit('feeds.mark_all_notifications_as.error', { args, error });
+        throw error;
+      }
+    });
+  }
+
+  async markNotificationActionAs(args: MarkNotificationActionAsByIdArgs): Promise<Notification>;
+  async markNotificationActionAs(args: MarkNotificationActionAsByInstanceArgs): Promise<Notification>;
```

Comment:
Let’s add specific method per action for better DX.

Reply:
Then we should do the same for the other functions as well, like the same way I did in the Notification class:
- markNotificationAsRead, markNotificationAsSeen, markNotificationAsUnseen, markNotificationAsUnread
- markNotificationPrimaryActionAsDone, markNotificationPrimaryActionAsPending, markNotificationSecondaryActionAsDone, markNotificationSecondaryActionAsPending

But then the question would be whether we should also emit separate events?